shortfin/shortfin/llm/impl/service_v1_cli.py

```diff
@@ -0,0 +1,91 @@
+
+import asyncio
+import argparse
+import numpy
+import sys
+
+from transformers import LlamaTokenizer
+
+from iree.runtime import (  # type: ignore
+    HalElementType,
+)
+
+from shortfin.framework.session import DeviceSession
+
+from shortfin.llm.attn_block_cache import (
+    create_attn_block_cache_module,
+    AttnBlockCache,
+)
+
+from shortfin.llm.config import (
+    CacheParams,
+    ModelParams,
+    ServiceParams,
+)
+
+from shortfin.llm.impl.service_v1 import GenerateServiceV1
+from shortfin.llm.service import GenerateRequest
+
+def setup(vmfb_path, config_path, gguf_path):
+    from iree.runtime._binding import disable_leak_checker  # type: ignore
+    model_params = ModelParams.load_json(config_path)
+
+    cache_params = CacheParams(
+        model=model_params, device_block_count=128, block_pos_stride=16)
+
+    disable_leak_checker()
+    session = DeviceSession(uri="local-task", queue_count=2)
+    attn_block_cache = AttnBlockCache(session, cache_params)
+
+    lms = session.create_module_set(model_params.module_name, context_count=1)
+    lms.load_io_module(gguf_path)
+    lms.load_vmfb(vmfb_path)
+    lms.add(create_attn_block_cache_module(attn_block_cache))
+    lms.initialize()
+
+    params = ServiceParams(cache=cache_params, model=model_params)
+    service = GenerateServiceV1(session=session, params=params, cache=attn_block_cache)
+    return service
+
+def map_buffer(value):
+    mapped = value.map()
+    return mapped.asarray(
+        value.shape,
+        HalElementType.map_to_dtype(value.element_type))
+
+async def main(argv):
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--tokenizer")
+    parser.add_argument("--config")
+    parser.add_argument("--vmfb")
+    parser.add_argument("--gguf")
```

Comment:
Document these? (They look pretty self-explanatory to me right now, but docstrings are easy enough to add and will help others in the future)

Reply:
done.